### ISSUE TITLE
Design of components for bullets, grenades and rockets

### ISSUE THREAD

**Issue by geneotech** [TeamHypersomnia/Hypersomnia#238]
damage will be renamed to **missile**.
grenade will be renamed to **explosive**. A flag will be introduced to determine whether it can be unpinned and thrown with just a hand.
a contact_explosive will be implemented by way of **explosive** + **missile**, removing the need for a separate component.

In the future, to resolve guilt/morality mechanics, new component will be introduced, named "sender" that will be accompanying all adversary objects (whether explosives, missiles or other kinds). 
But for now, fixtures shall henceforth possess an array of three optional entities with whom collision is to be filtered (e.g. the gun, the holder of the gun, the vehicle of the holder), initialized at the time of setting the missile in motion, and thus lifting one responsibility from **missile**.

If a missile_system (former name: damage_system) detects that a missile is due to be destroyed (whether due to contact or a lifetime constraint - already implemented for gun bullets), it shall also check for the existence of an explosive within that entity. If one is found, a detonation shall follow.

Thusly:
A gunshot round shall possess a _missile_.
A grenade shall possess an _explosive_.
A rocket shall possess both a _missile_ and an _explosive_.

**Comment by geneotech** [comment]
There is still a problem:
an explosive is currently in charge of another lifetime constraint (fields: when_released, when_explodes), a functionality that is already duplicated in _missile_.
Thus, a rocket possessing both a _missile_ and an _explosive_ needlessly holds data for two lifetime constraints.

Solution:
Let there be a separate component named _hand_fuse_. A hand fuse shall be in charge of letting a character initiate some delayed event that will be done upon that item. In case of a grenade, an explosion shall follow. There may be some other uses for it, and the fuse itself has no knowledge that it is attached to an explosive device or any other.

To summarize again:

A gunshot round shall possess a _missile_.
A grenade shall possess an _explosive_ and a _fuse_.
A rocket shall possess an _explosive_ and a _missile_.

**Comment by geneotech** [comment]
Now that I think of it, I will already just create a sender component that will serve as both the filter and the resolver of guilt or friendly fire. Otherwise, some data will be duplicated.

**Comment by geneotech** [comment]
eb81c73: Actually, we currently do have a field like this, and it is named _damage_charges_before_destruction_. Moreover, we actually need to ensure that we don't queue that entity twice, because coincidentally, there is also detonation involved. A missile shall by no means detonate twice.
